modules/RTC/TPCUtils.js

```diff
@@ -571,11 +670,90 @@ export class TPCUtils {
 
         // Implement the workaround only when all the encodings report the same resolution.
         if (allEqualEncodings(parameters.encodings)) {
-            const videoStreamEncodings = this._getVideoStreamEncodings(localVideoTrack.getVideoType());
+            const videoStreamEncodings = this._getVideoStreamEncodings(
+                localVideoTrack.getVideoType(),
+                this.pc.getConfiguredVideoCodec());
 
             parameters.encodings.forEach((encoding, idx) => {
                 encoding.scaleResolutionDownBy = videoStreamEncodings[idx].scaleResolutionDownBy;
             });
         }
     }
+
+    /**
+     * Updates the scalability mode on the encodings and adjusts the other parameters when the client is configured to
+     * use the scalability mode API.
+     *
+     * @param {Array<RTCRtpEncodingParameters>} encodings - The encodings as returned by the browser for the local
+     * video track.
+     * @param {CodecMimeType} codec - The codec that is in use.
+     * @param {number} maxHeight - The height of the outgoing resolution that needs to be configured.
+     * @param {JitsiLocalTrack} localVideoTrack - The local video track.
+     * @returns {void>}
+     */
+    updateEncodingsScalabilityMode(encodings, codec, maxHeight, localVideoTrack) {
+        if (!this.codecSettings[codec].scalabilityModeEnabled) {
+            return;
+        }
+
+        // Configure L1T3 mode on each of the encodings when H.264 is the selected codec.
```

Comment:
Why do we want that on every encoding? Does it make sense also for high layer?

Reply:
Yes, since H.264 supports only simulcast, we want all stream to be temporally scalable, even the stream with the highest resolution.